app/Command/Compile.hs

```diff
@@ -89,94 +84,66 @@ globWarningOnMisses warn = concatMapM globWithWarning
-  <> help "Display verbose error messages"
-
-noPrefix :: Parser Bool
-noPrefix = switch $
-     short 'p'
-  <> long "no-prefix"
-  <> help "Do not include comment header"
-
-jsonErrors :: Parser Bool
-jsonErrors = switch $
-     long "json-errors"
-  <> help "Print errors to stderr as JSON"
-sourceMaps :: Parser Bool
-sourceMaps = switch $
-     long "source-maps"
-  <> help "Generate source maps"
-
-dumpCoreFn :: Parser Bool
-dumpCoreFn = switch $
-     long "dump-corefn"
-  <> help "Dump the (functional) core representation of the compiled code at output/*/corefn.json"
-
-
-options :: Parser P.Options
-options = P.Options <$> noTco
-                    <*> noMagicDo
-                    <*> pure Nothing
-                    <*> noOpts
-                    <*> verboseErrors
+  <> Opts.showDefault
+  <> Opts.help "The output directory"
+
+comments :: Opts.Parser Bool
+comments = Opts.switch $
+     Opts.short 'c'
+  <> Opts.long "comments"
+  <> Opts.help "Include comments in the generated code"
+
+verboseErrors :: Opts.Parser Bool
+verboseErrors = Opts.switch $
+     Opts.short 'v'
+  <> Opts.long "verbose-errors"
+  <> Opts.help "Display verbose error messages"
+
+noPrefix :: Opts.Parser Bool
+noPrefix = Opts.switch $
+     Opts.short 'p'
+  <> Opts.long "no-prefix"
+  <> Opts.help "Do not include comment header"
+
+jsonErrors :: Opts.Parser Bool
+jsonErrors = Opts.switch $
+     Opts.long "json-errors"
+  <> Opts.help "Print errors to stderr as JSON"
+
+sourceMaps :: Opts.Parser Bool
+sourceMaps = Opts.switch $
+     Opts.long "source-maps"
+  <> Opts.help "Generate source maps"
+
+dumpCoreFn :: Opts.Parser Bool
+dumpCoreFn = Opts.switch $
+     Opts.long "dump-corefn"
+  <> Opts.help "Dump the (functional) core representation of the compiled code at output/*/corefn.json"
+
+options :: Opts.Parser P.Options
+options = P.Options <$> verboseErrors
                     <*> (not <$> comments)
                     <*> sourceMaps
                     <*> dumpCoreFn
 
-pscMakeOptions :: Parser PSCMakeOptions
+pscMakeOptions :: Opts.Parser PSCMakeOptions
 pscMakeOptions = PSCMakeOptions <$> many inputFile
                                 <*> outputDirectory
                                 <*> options
                                 <*> (not <$> noPrefix)
                                 <*> jsonErrors
 
-main :: IO ()
-main = do
-  hSetEncoding stdout utf8
-  hSetEncoding stderr utf8
-  execParser opts >>= compile
-  where
-  opts        = info (version <*> helper <*> pscMakeOptions) infoModList
-  infoModList = fullDesc <> headerInfo <> footerInfo
-  headerInfo  = header   "psc - Compiles PureScript to Javascript"
-  footerInfo  = footer $ "psc " ++ showVersion Paths.version
-
-  version :: Parser (a -> a)
-  version = abortOption (InfoMsg (showVersion Paths.version)) $ long "version" <> help "Show the version number" <> hidden
+command :: Opts.Parser (IO ())
+command = compile <$> (version <*> Opts.helper <*> pscMakeOptions) where
```

Comment:
I'm not sure the sub commands need to be able to respond to `--version`. Having `psc --version` work is enough IMO, we don't also need to support `psc build --version` etc.